### ISSUE TITLE
Incorrect size calculation

### ISSUE THREAD

**Issue by ciriousjoker** [alexmamo/firebase-extensions#1]
```ts
const data = {
  date: Timestamp.now(),
  value: 999,
};
const arr = Array(38000).fill(data);
await firestore().collection("test").doc("1").set({
  1: arr,
});
```

This is reportedly 3116057 bytes bit, which makes no sense because it fits within the 1,048,576 bytes limit. When using 40k instead of 38k, it barely doesn't fit:

> Document 'projects/.../databases/(default)/documents/test/1' cannot be written because its size (1,080,057 bytes) exceeds the maximum allowed size of 1,048,576 bytes.

**Comment by alexmamo** [comment]
I found the mistake. It turned out that [documentation](https://firebase.google.com/docs/firestore/storage-size#document-size) doesn't correctly explain how to calculate the size of a Map field. There is no need to add the document name size and those additional 32 bytes. I'll update the Extension.

**Comment by alexmamo** [comment]
I found the mistake. It turned out that [documentation](https://firebase.google.com/docs/firestore/storage-size#document-size) doesn't correctly explain how to calculate the size of a Map field. There is no need to add the document name size and those additional 32 bytes. Here are some calculations with 38k and 40k maps:

documentNameSize = (4 + 1) + (1 + 1) + 16 = 23
arrayNameSize = (1 + 1) = 2
arrayValueSize = {[(4 + 1) + 8] + [(5 + 1) + 8]} = 13 + 14 = 27 * 38.000 = 1.026.000
documentAdditionalBytes = 32
--- total---
23 + 2 + 1.026.000 + 32 = 1.026.057 (correct < 1.048.576)
-------------------------------------------------------------------------------------------
documentNameSize = (4 + 1) + (1 + 1) + 16 = 23
arrayNameSize = (1 + 1) = 2
arrayValueSize = {[(4 + 1) + 8] + [(5 + 1) + 8]} = 13 + 14 = 27 * 40.000 = 1.080.000
documentAdditionalBytes = 32
--- total---
23 + 2 + 1.080.000 + 32 = 1.080.057 (correct > 1.048.576)

I'll update the Extension.

**Comment by ciriousjoker** [comment]
Ah yes I was kind of scared that maps were so expensive. A map with a single entry apparently cost 56 bytes!

**Comment by alexmamo** [comment]
Yes, that's what the docs say but actually, it isn't so expensive. I'll let you know when the new version is available.

**Comment by alexmamo** [comment]
Hey Philipp. They just have approved my updated version. So it should work now. I will close this issue. However, if you find anything else that doesn't work correctly, please let me know. Thanks again.
